framework/wazuh/engine/queries.py

```diff
@@ -0,0 +1,420 @@
+
+        Args:
+            params: The parameters used for the transformation.
+            data: The data to be transformed.
+
+        Returns:
+            bool: True if the transformation can be applied, False otherwise.
+
+        Raises:
+            NotImplementedError: This method must be implemented in the derived classes.
+        """
+        raise NotImplementedError
+
+
+class FieldSelector(BaseQuery):
+    """Transformation to select specific fields from the data."""
+
+    def __init__(self, params: Dict[str, Any]):
+        self.select = params['select']
+
+    @staticmethod
+    def can_i_run(params: Dict[str, Any], data: Any) -> bool:
+        if 'select' in params and params['select'] and isinstance(data, list):
+            return True
+        return False
+
+    def apply_transformation(self, data: List[Dict[str, Any]]):
+        """Apply the field selection transformation to the data.
+
+        Args:
+            data: The data to be transformed.
+
+        Returns:
+            The transformed data.
+
+        Note:
+            If the data is a list of dictionaries, the transformation is applied to each dictionary.
+        """
+        # Create an empty list if the data is a list, otherwise create an empty dictionary
+        transformed_data = [] if isinstance(data, list) else {}
+
+        # Remove leading/trailing spaces and split the selected fields based on comma separation
+        selected_fields = [name.strip() for name in self.select.split(",")]
+
+        # If the data is a list, iterate over each element
+        for element in data:
+            selected_elements = self._select_fields(selected_fields, element)
+            transformed_data.append(selected_elements)
+
+        # Return the transformed data
+        return transformed_data
+
+    @staticmethod
+    def _select_fields(selected_fields: List[str], data: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Selects and extracts specified fields from the given data dictionary.
+
+        Args:
+            selected_fields (List[str]): A list of field names to be selected.
+            data (Dict[str, Any]): The data dictionary to select fields from.
+
+        Returns:
+            Dict[str, Any]: A dictionary containing the selected fields and their values.
+
+        Example:
+            selected_fields = ['name', 'age', 'address.city']
+            data = {'name': 'John', 'age': 25, 'address': {'street': '123 Main St', 'city': 'New York'}}
+            result = MyClass._select_fields(selected_fields, data)
+            print(result)
+            # Output: {'name': 'John', 'age': 25, 'address': {'city': 'New York'}}
+        """
+
+        # Create an empty dictionary to store the selected fields for the current element
+        selected_element = {}
+        for name in selected_fields:
+            # Get the value of the selected field for the current element
+            value = FieldSelector._get_nested_value_select(data, name)
```

Comment:
In my opinion, this should be called using `self`. So, `_select_fields` shouldn't be a static method just an instance one.

Reply:
https://github.com/wazuh/wazuh/blob/92f0a22c401a9cdc766321dfd3229823ae849dbb/framework/wazuh/core/engine/queries.py#L113